WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
         return
     }
-    Write-Host "WSL is already installed. Skipping WSL installation."
+    elseif($distrostatus -eq -1)
+    {
+        Write-Host "WSL distro $distroName is not installed. Installing WSL distro $distroName..."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        Write-Success "WSL distro $distroName is installed. Run Initialize-WSLBlobNFS to setup WSL environment for WSLBlobNFS usage."
+
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
+        return
+    }
+
     return
 }
 
-function Initialize-WSLBlobNFS
+
+function Initialize-WSLBlobNFS-Internal
 {
-    # Most of the commands require admin privileges. Hence, we need to run the script as admin.
-    $linuxusername = "root"
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $false)]$Force=$false
+    )
 
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    Intall-WSLBlobNFS-Internal
+    if($LASTEXITCODE -ne 0)
     {
-        Write-Host "WSL is not installed. Installing WSL..."
-
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
         return
     }
 
-    # Run the script with setupwslenv argument
-    # To-do: Automate to start this on startup.
+    $initialized = $false
+    if($Force)
+    {
+        Write-Warning "Force initializing WSL environment for WSLBlobNFS usage."
+    }
+    else
+    {
+        Write-Debug "Initializing WSL environment for your WSLBlobNFS usage."
+    }
 
-    Write-Host "Installing Ubuntu-22.04..."
-    wsl --install -d Ubuntu-22.04
+    # # Add the modulePathForLinux, which hold the path of module, to WSLENV to help in copying linux scripts to wsl distro.
+    # [Environment]::SetEnvironmentVariable("modulePathForLinux", $PSScriptRoot)
+    # $p = [Environment]::GetEnvironmentVariable("WSLENV")
+    # $p += ":modulePathForLinux/p"
+    # [Environment]::SetEnvironmentVariable("WSLENV",$p)
 
-    # ask user if the username was setup
-    $usernameSetup = Read-Host "Was the wsl distro username setup? (y/n)"
-    if($usernameSetup -ne "y")
+    $moduleVersionOnWin = Get-ModuleVersion
+    Write-Debug "Module version on Windows: $moduleVersionOnWin"
+    $moduleVersionOnWSL = Execute-WSL "cat $linuxScriptsPath/$versionFileName 2>/dev/null"
+    Write-Debug "Module version on WSL: $moduleVersionOnWSL"
+    if(($LASTEXITCODE -eq 0) -and ($moduleVersionOnWSL.Trim() -eq $moduleVersionOnWin) -and !$Force)
```

Comment:
this LastExitCode should correspond to Write-Debug, isn't it?
Also, as per my comment above, if we don't copy the files and use them from just one place, we don't need this copying check.

Reply:
no, correct debug and out commands are affecting the lastexitcode.